File: apps/backend/core/supabase_client.py
# core/supabase_client.py
import os
from supabase import create_client

SUPABASE_URL = os.getenv("SUPABASE_URL")

# Cliente normal (anon key)
SUPABASE_ANON_KEY = os.getenv("SUPABASE_ANON_KEY")
supabase = create_client(SUPABASE_URL, SUPABASE_ANON_KEY)

# Cliente admin (service role key, nunca exponer al frontend)
SUPABASE_SERVICE_ROLE_KEY = os.getenv("SUPABASE_SERVICE_ROLE_KEY")
supabase_admin = create_client(SUPABASE_URL, SUPABASE_SERVICE_ROLE_KEY)

File: apps/web/src/lib/supabaseClient.ts
// src/lib/supabaseClient.ts
import { createClient } from "@supabase/supabase-js";

// ⚠️ Lo ideal es que uses variables de entorno en tu proyecto frontend (.env)
// nunca dejes la KEY hardcodeada en el repo público.

const supabaseUrl = import.meta.env.VITE_SUPABASE_URL || process.env.SUPABASE_URL!;
const supabaseAnonKey = import.meta.env.VITE_SUPABASE_KEY || process.env.SUPABASE_KEY!;

// Inicializa cliente
export const supabase = createClient(supabaseUrl, supabaseAnonKey);
